gcp_variant_transforms/beam_io/vcf_parser.py

```diff
@@ -309,13 +360,26 @@ def _process_header_lines(self, header_lines):
       # We need to keep the last line of the header from the file because it
       # contains the sample IDs, which is unique per file.
       header_lines = self._representative_header_lines + header_lines[-1:]
-    self._init_with_header(header_lines)
+    if isinstance(self, PySamParser):
```

Comment:
Same here: a description of why this header munging is needed for the pysam case is helpful